### ISSUE TITLE
Listen Section doesn't forward the same 503 response on queue timeout

### ISSUE THREAD

**Issue by madhanrajrm** [haproxy/haproxy#2659]
### Detailed Description of the Problem

I have proxy section  , frontend  and backend. 

When my requests reaches backend where i have queue timeout set there . If request hits queue timeout i see backend sends back 503 properly . But when frontend forwards the same to listen section  i see listen section responds back with 502 error instead of 503. 

So i would want listen section too forward the same 503 response instead of 502. I feel this shouldn't work this way or does proxy always rewrites the response into backend ? 

### Expected Behavior

So i would want listen section too forward the same 503 response instead of 502.

### Steps to Reproduce the Behavior

configure an queue timeout with around  1 sec and pump more requests to backend . when it hits timeout though backend sends back 503 listen section will send 502 back .

### What is your configuration?

```haproxy
global
  maxconn 32768
  nbthread 4
  maxsslrate 500
  maxsslconn 18000
  tune.bufsize 32768
  tune.ssl.default-dh-param 2048
  tune.ssl.lifetime 3600
  log /dev/log local5

  stats socket ipv4@127.0.0.1:9999 level admin
  stats socket /var/run/haproxy.sock mode 666 level admin
  stats timeout 2m

  ssl-default-bind-ciphers ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384
  ssl-default-bind-ciphersuites TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256
  ssl-default-bind-options prefer-client-ciphers ssl-min-ver TLSv1.2  ssl-max-ver TLSv1.3
  ssl-default-bind-curves P-256:P-384:P-521

defaults
  log     global
  mode    http
  option  log-health-checks
  option  log-separate-errors
  option  dontlognull
  log-format "%tr %ci:%cp %fp/%fc/%Th/%Tq/%rc %ft %b/%s %TR/%Tw/%Tc/%Tr/%Ta %ST %B %CC %CS %tsc %ac/%fc/%bc/%sc/%rc %sq/%bq %hr %hs %{+Q}HP"
  option  splice-auto
  option  socket-stats
  option  redispatch
  maxconn 5000
  timeout connect 3s
  timeout client 300s
  timeout server 600s
  timeout client-fin 1s
  timeout server-fin 1s

listen ssl
  bind 0.0.0.0:8443 tfo ssl crt /usr/HAProxy.pem npn http/1.1
  maxconn 30000
  rate-limit sessions 200
  server http 127.0.0.1:84 send-proxy-v2

frontend http
  bind 127.0.0.1:84 tfo accept-proxy
  acl is_ssl fc_rcvd_proxy
  acl isport res.hdr(location) -m beg http://localhost:
  use_backend uds-tomcat

backend uds-tomcat
  balance roundrobin
  timeout queue 1s
  option nolinger
  server s1 127.0.0.1:8080 maxconn 1
```

### Output of `haproxy -vv`

```plain
[root@ccm ~]# /usr/local/cm/bin/haproxy -vvv
HA-Proxy version 2.2.32-4081d5a 2023/12/19 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2025.
Known bugs: http://www.haproxy.org/bugs/bugs-2.2.32.html
Running on: Linux 4.18.0-513.11.1.el8_9.x86_64 #1 SMP Wed Jan 17 02:00:40 EST 2024 x86_64
Build options :
  TARGET  = custom
  CPU     = generic
  CC      = gcc
  CFLAGS  = -m64 -march=x86-64 -O2 -g -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-stringop-overflow -Wno-cast-function-type -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_EPOLL=1 USE_THREAD=1 USE_LINUX_SPLICE=1 USE_OPENSSL=1 USE_CPU_AFFINITY=1
  DEBUG   =

Feature list : -51DEGREES -ACCEPT4 -BACKTRACE -CLOSEFROM +CPU_AFFINITY -CRYPT_H -DEVICEATLAS -DL +EPOLL -EVPORTS -FUTEX -GETADDRINFO -KQUEUE -LIBCRYPT +LINUX_SPLICE -LINUX_TPROXY -LUA -NETFILTER -NS -OBSOLETE_LINKER +OPENSSL -PCRE -PCRE2 -PCRE2_JIT -PCRE_JIT +POLL -PRCTL -PRIVATE_CACHE -PTHREAD_PSHARED -RT -SLZ -STATIC_PCRE -STATIC_PCRE2 -SYSTEMD -TFO +THREAD -THREAD_DUMP -TPROXY -WURFL -ZLIB

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=8).
Built with OpenSSL version : OpenSSL 1.1.1w.7.2.555
Running on OpenSSL version : OpenSSL 1.1.1w.7.2.555
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built without compression support (neither USE_ZLIB nor USE_SLZ are set).
Compression algorithms supported : identity("identity")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built without PCRE or PCRE2 support (using libc's regex instead)
Encrypted password support via crypt(3): no
Built with gcc compiler version 8.5.0 20210514 (Red Hat 8.5.0-20)

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
            fcgi : mode=HTTP       side=BE        mux=FCGI
       <default> : mode=HTTP       side=FE|BE     mux=H1
              h2 : mode=HTTP       side=FE|BE     mux=H2
       <default> : mode=TCP        side=FE|BE     mux=PASS

Available services : none

Available filters :
        [SPOE] spoe
        [COMP] compression
        [TRACE] trace
        [CACHE] cache
        [FCGI] fcgi-app
```

### Additional Information

 this in haproxy log for one of the requests.

Jul 31 00:53:34: 31/Jul/2024:00:53:14.420 10.89.109.191:46978 8443/65/0/0/0 http uds-tomat/<NOSRV> 0/1001/-1/-1/100 **503** 222 - - sQ-- 173/65/9/0/0 0/349 {220|15} "/ann/user/User1101202/remoteDestinations"
Jul 31 00:53:34: 31/Jul/2024:00:53:14.419 10.89.109.191:46978 8443/62/4/4/0 ssl~ ssl/http 0/0/0/-1/1004 **502** 214 - - SH-- 173/62/36/36/0 0/0 "/ann/user/User1101202/remoteDestinations"

**Comment by einval22** [comment]
Hi  madhanrajrm !

Thanks for reporting this.

This issue seems to be fixed in 2.2.33 bug fix version, which is available [here](http://www.haproxy.org/download/2.2/src/haproxy-2.2.33.tar.gz), and in a latest 3.0.3.

This patch fixes this problem:
https://git.haproxy.org/?p=haproxy-2.2.git;a=commit;h=5fb39e2fab37d22f56694162c77e13ecb40c98f2

Could you please update and report us the results.

**Comment by madhanrajrm** [comment]
Thanks @vkssv  i see it is fixed in 2.2.33
